_posts/2024-02-27-ba_importer.md

```diff
@@ -0,0 +1,119 @@
+---
+layout: post
+title: "[CVE-2024-25843] Improper neutralization of SQL parameter in Buy Addons - Import/Update Bulk Product from any Csv/Excel File Pro module for PrestaShop"
+categories: modules
+author:
+- TouchWeb.fr
+- 202 Ecommerce
+- Friends-Of-Presta.org
+meta: "CVE,PrestaShop,ba_importer"
+severity: "critical (9.8)"
+---
+
+In the module "Import/Update Bulk Product from any Csv/Excel File Pro" (ba_importer) up to version 1.1.28 from Buy Addons for PrestaShop, a guest can perform SQL injection in affected versions.
+
+## Summary
+
+* **CVE ID**: [CVE-2024-25843]
+* **Published at**: 2024-02-27
+* **Platform**: PrestaShop
+* **Product**: ba_importer
+* **Impacted release**: <= 1.1.28 (1.1.29 fixed the vulnerability)
+* **Product author**: Buy Addons
+* **Weakness**: [CWE-89](https://cwe.mitre.org/data/definitions/89.html)
+* **Severity**: critical (9.8)
+
+## Description
+
+The method `ba_importerAjaxSettingModuleFrontController::run()` has sensitive SQL calls that can be executed with a trivial http call and exploited to forge a SQL injection.
+
+**WARNING** : This exploit is actively used to deploy a webskimmer to massively steal credit cards.
+
+This exploit uses a PrestaShop front controller and most attackers can conceal the module controller's path during the exploit, so you will never know within your conventional frontend logs that it exploits this vulnerability. **You will only see "POST /" inside your conventional frontend logs.** Activating the AuditEngine of mod_security (or similar) is the only way to get data to confirm this exploit.
+
+Note : the author has deleted from its module the file which have been suffering from the critical vulnerability for years, BUT did not set them to be "auto-deleted" during upgrades. Therefore, there are likely merchants out there with older versions who have updated their modules, thinking they are safe. However, there is nothing safe about this, since past upgrades did not auto-delete the implicated files. To ensure everyone has a "safe version", we decided to mark all versions up to X.X.X as impacted by this issue.
```

Comment:
the author has deleted the file from his module wich....

Reply:
there is nothing safe about this => there is nothing safe about that

Reply:
Merci Arnaud pour tes retours. C'est corrigé !